### PR TITLE
Adds Support type and method

### PR DESCRIPTION
**Pull request by sobolevn**
Reviews are welcome!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ And it easy to extend this typeclass with your own classes as well:
 ```
 
 That's how simple, safe, and powerful typeclasses are!
-Make sure to [check out our docs](https://github.com/dry-python/classes) to learn more.
+Make sure to [check out our full docs](https://github.com/dry-python/classes) to learn more.
```
Comment (thepabloaguilar) [inline README.md:142]: Just a question, why `full`? Hahahaha

**Review by thepabloaguilar — CHANGES_REQUESTED** [review]

File: classes/_typeclass.py [hunk 2]
```diff
@@ -318,3 +177,256 @@ def typeclass(
...
+        is_protocol: bool = False,
+    ):
+        """
+        We use this method to store implementation for each specific type.
+
+        The only setting we provide is ``is_protocol`` which is required
+        when passing protocols.
+        That's why we also have this ugly ``@overload`` cases.
+        Otherwise, ``Protocol`` instances
+        would not match ``Type[_InstanceType]`` type due to ``mypy`` rules.
```
[the remark this answers is not in the document]
Comment (sobolevn) [reply classes/_typeclass.py:423]: This is actually a great feature. It does not allow to use `some_typeclass.instance(Sized)` without `is_protocol=True`.

File: README.md (hunk 1 above)
Comment (sobolevn) [reply README.md:142]: I had a warning that we have two similar anchors!

**Review by thepabloaguilar — APPROVED** [review]
LGTM

**Comment by sobolevn** [comment]
Great!

**Pull request merged**